core/cli/cli.js

```diff
@@ -1,10 +1,47 @@
 #!/usr/bin/env node
 
-process.on('unhandledRejection', err => {throw err});
+const program = require('commander');
+const pkg = require('./package.json');
+const fs = require('fs-extra');
+const path = require('path');
+let argValue;
+
+program
+    .version(pkg.version, '-v, --version')
+    .command('init <project-name>')
+    .description('creates new project scaffolding')
+    .action( (project) => {
+        argValue = 'init';
+        createScaffold(project);
+    });
+
+program
+    .command('start')
+    .description('starts dev server on localhost')
+    .action( () => {
+        argValue = 'start';
+        start();
+    });
+
+program
+    .arguments('<arg>')
+    .action((arg) => {
+        argValue = arg;
+        console.error(`Sorry, ${arg} is not a valid command!  Please use -h or --help for valid commands.`);
+    });
+
+program.parse(process.argv);
+
+if (typeof argValue === 'undefined') {
+    console.error('No command given!  Please use -h or --help for valid commands.');
+    process.exit(1);
+}
 
-run();
+//process.on('unhandledRejection', err => {throw err});
```

Comment:
why comment this away?

Reply:
Oh yea!  I wanted to ask you about that.  Still kind of learning about the way processes work and wasn't sure if we needed this with the error handling I passed through commander, but actually with the statements above, I think I see what's going on.  So that's basically just an event listener and if there's an error, it will stop the process and throw an error, correct?

Just out of curiosity, instead of throwing the error, should we just console.log it to make it look cleaner?  we could do something like:

```javascript
process.on('unhandledRejection', err => {
  console.log(err);
  process.exit(1);
}
```

If you prefer to throw, though, I'll follow your lead :)

Reply:
> So that's basically just an event listener and if there's an error, it will stop the process and throw an error, correct?

Yes. More precisely it is rejected promises that are listened for. But I guess that's what you were thinking.

And yes we could `console.log(err); process.exit(1);` instead. I just find that `throw err` expresses our intention slightly better. But that's nitpicking and either way is fine to me.